### PR TITLE
disk_block_cache: Correctly track the disk usage for existing block caches

### PR DESCRIPTION
**Pull request by jzila**

### CODE REVIEW

File: libkbfs/disk_block_cache.go [hunk 1]
```diff
@@ -525,10 +527,6 @@ func (cache *DiskBlockCacheStandard) UpdateLRUTime(ctx context.Context,
 		return DiskCacheStartingError{"UpdateLRUTime"}
 	}
 	var md diskBlockCacheMetadata
-	defer func() {
-		cache.log.CDebugf(ctx, "Cache UpdateLRUTime id=%s entrySize=%d "+
-			"err=%+v", blockID, md.BlockSize, err)
-	}()
```
Comment (jzila) [inline libkbfs/disk_block_cache.go]: This was waaaay too noisy in the log.

File: libkbfs/disk_block_cache.go [hunk 2]
```diff
@@ -165,8 +162,13 @@ func newDiskBlockCacheStandardFromStorage(config diskBlockCacheConfig,
 				"block counts from DB: %+v", err)
 			return
 		}
-		// Only enable compaction after the block counts have been synced.
-		compactCh <- struct{}{}
+		diskLimiter := cache.config.DiskLimiter()
+		if diskLimiter != nil {
+			// Only enable compaction after the block counts have been synced.
```
Comment (strib) [inline libkbfs/disk_block_cache.go]: Compaction?  Isn't this enabling the disk limiter for the cache?

File: libkbfs/disk_block_cache.go [hunk 3]
```diff
@@ -667,9 +665,17 @@ func (*DiskBlockCacheStandard) getRandomBlockID(numElements,
 	return kbfsblock.MakeRandomIDInRange(0, pivot)
 }
 
+// evictSomeBlocks tries to evict <numBlocks> blocks from the cache. If
```
Comment (strib) [inline libkbfs/disk_block_cache.go]: Can we standardize on markdown ticks for variable names in comments?  i.e. \`numBlocks\`.  (And \`blockIDs\` on the next line.)

File: libkbfs/disk_block_cache.go [hunk 4]
```diff
@@ -667,9 +665,17 @@ func (*DiskBlockCacheStandard) getRandomBlockID(numElements,
...
+// blockIDs doesn't have enough blocks, we evict them all and report how many
+// we evicted.
 func (cache *DiskBlockCacheStandard) evictSomeBlocks(ctx context.Context,
 	numBlocks int, blockIDs blockIDsByTime) (numRemoved int, sizeRemoved int64,
 	err error) {
+	defer func() {
+		cache.log.CDebugf(ctx, "Cache evictSomeBlocks numBlocksRequested=%d, "+
```
Comment (strib) [inline libkbfs/disk_block_cache.go]: Also print out `err`.

**Comment by jzila** [comment]
@strib: yes, this'll fix everyone. It'll evict on `Put`s, distributed over many of them, and none of them will succeed until enough blocks are evicted. Since `Put`s happen asynchronously, it should be completely transparent.

**Comment by strib** [comment]
Sweet thanks!

**Pull request merged**